components/shared_data_model/ci/pip-compile.sh

```diff
@@ -10,6 +10,4 @@ run () {
 }
 
 # Update the compiled requirements files
-cd requirements
-run pip-compile --allow-unsafe --generate-hashes --quiet --resolver=backtracking --upgrade requirements-base.in
-run pip-compile --allow-unsafe --generate-hashes --quiet --resolver=backtracking --upgrade requirements-dev.in
+run pip-compile --allow-unsafe --generate-hashes --quiet --resolver=backtracking --upgrade --extra dev --output-file requirements/requirements-dev.txt pyproject.toml
```

Comment:
Is there no regular install because of packaging?

Reply:
Yes. We may want to look into how we install the shared components in the Docker images of the components because we use editable installs there:

```Dockerfile
COPY collector/requirements/requirements-internal.txt /requirements-internal.txt
RUN pip install --no-cache-dir -r /requirements-internal.txt
```

The requirements-internal.txt reads:

```
-e ../shared_data_model
```

But that's outside the scope of the PR I'd say.